### ISSUE TITLE
Create a new GrSimBackend to output SensorProto and uses ProtoMulticastListener

### ISSUE THREAD

**Issue by jonathanlew** [UBC-Thunderbots/Software#1281]
### Description of the task
Create a new GrSimBackend to output SensorProto and use ProtoMulticastListener instead of NetworkClient. This should be easily swapped in to integrate things in #1135

Note that it must implement the following functionality:
```
void NetworkClient::publishVisionDataWrapper(SSL_WrapperPacket packet)
{
    // We analyze the first 60 packets we receive to find the "real" starting time,
    // because GrSim will sometimes give us garbage packets.
    // The real starting time is the smaller value of the ones we receive
    if (initial_packet_count < 60)
    {
        initial_packet_count++;
        if (packet.has_detection() &&
            packet.detection().t_capture() < last_valid_t_capture)
        {
            last_valid_t_capture = packet.detection().t_capture();
        }
    }
    else
    {
        // We pass all packets without a detection to the logic (since they are likely
        // geometry packet). Packets with detection timestamps are compared to the last
        // valid timestamp to make sure they are close enough before the data is passed
        // along. This ensures we ignore any of the garbage packets grsim sends that
        // are thousands of seconds in the future.
        if (!packet.has_detection())
        {
            publishVisionData(packet);
        }
        else if (std::fabs(packet.detection().t_capture() - last_valid_t_capture) < 100)
        {
            last_valid_t_capture = packet.detection().t_capture();
            publishVisionData(packet);
        }
    }
}
```

### Acceptance criteria

- [ ] Create a new GrSimBackend to output SensorProto and use ProtoMulticastListener instead of NetworkClient 
- [ ] Testing

### Blocked By
 #1283 #1280

**Comment by MathewMacDougall** [comment]
Just leaving a comment to mark this as lower priority, as hopefully we can just drop support for grSim with planned Simulator upgrades.
